cybox/common/vocabs.py

```diff
@@ -99,43 +157,1368 @@ def to_dict(self):
         # the value.
-        vocab_str.value = denormalize_from_xml(vocab_obj.valueOf_,
-                                               vocab_str.delimiter)
+        return_obj.value = denormalize_from_xml(
+            value=vocab_obj.valueOf_,
+            delimiter=return_obj.delimiter
+        )
 
-        return vocab_str
+        return return_obj
 
     @classmethod
-    def from_dict(cls, vocab_dict):
+    def from_dict(cls, vocab_dict, return_obj=None):
         if not vocab_dict:
             return None
 
-        vocab_str = cls()
-        # xsi_type should be set automatically by the class's constructor.
+        if not return_obj:
+            if isinstance(vocab_dict, dict):
+                klass = VocabString.lookup_class(vocab_dict.get('xsi:type'))
+                return_obj = klass()
+            else:
+                return_obj = cls()
 
         # In case this is a "plain" string, just set it.
         if not isinstance(vocab_dict, dict):
-            vocab_str.value = vocab_dict
+            return_obj.value = vocab_dict
         else:
-            vocab_str.xsi_type = vocab_dict.get('xsi:type', cls._XSI_TYPE)
-            vocab_str.value = vocab_dict.get('value')
-            vocab_str.vocab_name = vocab_dict.get('vocab_name')
-            vocab_str.vocab_reference = vocab_dict.get('vocab_reference')
+            return_obj.xsi_type = vocab_dict.get('xsi:type', cls._XSI_TYPE)
+            return_obj.value = vocab_dict.get('value')
+            return_obj.vocab_name = vocab_dict.get('vocab_name')
+            return_obj.vocab_reference = vocab_dict.get('vocab_reference')
+
+            PatternFieldGroup.from_dict(vocab_dict, return_obj)
+
+        return return_obj
+
+
+class EventType(VocabString):
+    _namespace = 'http://cybox.mitre.org/default_vocabularies-2'
+    _XSI_TYPE = 'cyboxVocabs:EventTypeVocab-1.0.1'
+    _VOCAB_VERSION = '1.0.1'
+    _ALLOWED_VALUES = (
+        'File Ops (CRUD)',
+        'Registry Ops',
+        'Memory Ops',
+        'Process Mgt',
+        'Thread Mgt',
+        'Service Mgt',
+        'Session Mgt',
+        'API Calls',
+        'Port Scan',
+        'IP Ops',
+        'DNS Lookup Ops',
+        'Socket Ops',
+        'IPC',
+        'Configuration Management',
+        'User/Password Mgt',
+        'Account Ops (App Layer)',
+        'HTTP Traffic',
+        'App Layer Traffic',
+        'Packet Traffic',
+        'Data Flow',
+        'Anomaly Events',
+        'Technical Compliance',
+        'Procedural Compliance',
+        'GUI/KVM',
+        'Autorun',
+        'USB/Media Detection',
+        'SQL',
+        'DHCP',
+        'Redirection',
+        'Authentication Ops',
+        'Authorization (ACL)',
+        'Privilege Ops',
+        'Basic System Ops',
+        'Signature Detection',
+        'Auto-update Ops',
+        'Application Logic',
+        'Email Ops',
+    )
+    TERM_APP_LAYER_TRAFFIC = 'App Layer Traffic'
```

Comment:
These should be sorted, too. I can change it pretty quickly if we agree that's what we want to do.